EngineLayer/NonSpecificEnzymeSearch/NonSpecificEnzymeEngine.cs

```diff
@@ -0,0 +1,334 @@
+                    if (new_progress > old_progress)
+                    {
+                        ReportProgress(new ProgressEventArgs(new_progress, "In modern search loop", nestedIds));
+                        old_progress = new_progress;
+                    }
+                }
+            });
+            return new SearchResults(newPsms, this);
+        }
+
+        private double Accepts(double scanPrecursorMass, CompactPeptide peptide, PpmTolerance precursorTolerance, Protease protease)
+        {
+            //all masses in N and CTerminalMasses are b-ion masses, which are one water away from a full peptide
+            int localminPeptideLength = minPeptideLength ?? 0;
+            if (protease.Name.Equals("singleN"))
+            {
+                for (int i = localminPeptideLength; i < peptide.NTerminalMasses.Count(); i++)
+                {
+                    double theoMass = peptide.NTerminalMasses[i] + waterMonoisotopicMass;
+                    if (Math.Abs((scanPrecursorMass - theoMass) / (theoMass) * 1e6) < precursorTolerance.Value)
+                    {
+                        return theoMass;
+                    }
+                }
+
+            }
+            else//if singleC
+            {
+                for (int i = localminPeptideLength; i < peptide.CTerminalMasses.Count(); i++)
+                {
+                    double theoMass = peptide.CTerminalMasses[i] + waterMonoisotopicMass;
+                    if (Math.Abs((scanPrecursorMass - theoMass) / (theoMass) * 1e6) < precursorTolerance.Value)
+                    {
+                        return theoMass;
+                    }
+                }
+            }
+            //if the theoretical and experimental have the same mass
+            if (peptide.NTerminalMasses.Count() > localminPeptideLength)
+            {
+                double totalMass = peptide.MonoisotopicMassIncludingFixedMods;// + Constants.protonMass;
+                if (Math.Abs((scanPrecursorMass - totalMass) / (totalMass) * 1e6) < precursorTolerance.Value)
+                {
+                    return totalMass;
+                }
+            }
+            return 0;
+        }
+
+        private void CalculatePeptideScores(IMsDataScan<IMzSpectrum<IMzPeak>> spectrum, double[] peptideScores, double thePrecursorMass, PpmTolerance precursorTolerance)
+        {
+            for (int i = 0; i < spectrum.MassSpectrum.Size; i++)
+            {
+                var theAdd = 1 + spectrum.MassSpectrum[i].Intensity / spectrum.TotalIonCurrent;
+                var experimentalPeakInDaltons = spectrum.MassSpectrum[i].Mz - Constants.protonMass;
+                GeneratePeptideScores(theAdd, experimentalPeakInDaltons, peptideScores, fragmentTolerance);
+            }
+            if (addCompIons)
+            {
+                List<IMzPeak> experimentalPeaks = new List<IMzPeak>();
+                //If HCD
+                if (lp.Contains(ProductType.B) | lp.Contains(ProductType.Y))
```

Comment:
That implies it can only be one or the other, though. For EThcD, they can't be conditioned like that.